src/dynamics/implicit.jl

```diff
@@ -64,6 +64,58 @@ function implicit_correction!(
     g = model.planet.gravity                    # gravitational acceleration [m/s²]
     ξ = implicit.time_step                      # new implicit timestep ξ = α*dt = 2αΔt (for leapfrog)
 
+    # Get precomputed l_indices from the spectrum
+    l_indices = div_tend.spectrum.l_indices
+
+    # GPU kernel launch
+    arch = architecture(div_tend)
+    launch!(arch, SpectralWorkOrder, size(div_tend), implicit_shallow_water_kernel!,
+            div_tend, pres_tend, div_old, div_new, pres_old, pres_new, l_indices, H, g, ξ)
+end
+
+@kernel inbounds=true function implicit_shallow_water_kernel!(
+    div_tend, pres_tend, div_old, div_new, pres_old, pres_new, l_indices,
+    @Const(H), @Const(g), @Const(ξ)
+)
+    I = @index(Global, Cartesian)
```

Comment:
This could be `lm, k = @index(Global, NTuple)` because the code below actually hardcodes it to 3D (lm x k) anyway?